python/pyspark/broadcast.py

```diff
@@ -177,28 +201,28 @@ def destroy(self, blocking=False):
         self._jbroadcast.destroy(blocking)
         os.unlink(self._path)
 
-    def __reduce__(self):
+    def __reduce__(self) -> Tuple[Callable[[int], T], Tuple[int]]:
         if self._jbroadcast is None:
             raise RuntimeError("Broadcast can only be serialized in driver")
-        self._pickle_registry.add(self)
-        return _from_id, (self._jbroadcast.id(),)
+        cast(Any, self._pickle_registry).add(self)
+        return cast(Tuple[Callable[[int], T], Tuple[int]], (_from_id, (self._jbroadcast.id(),)))
 
 
 class BroadcastPickleRegistry(threading.local):
     """ Thread-local registry for broadcast variables that have been pickled
     """
 
-    def __init__(self):
+    def __init__(self) -> None:
         self.__dict__.setdefault("_registry", set())
 
-    def __iter__(self):
+    def __iter__(self) -> Generator[Broadcast, None, None]:
```

Comment:
That differs from `broadcast.pyi` because we receive a `Generator` with `yield`

Reply:
I guess we can use `Iterator` instead of `Generator`.